<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="generator" content="quarto-1.7.32">
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">
<title>Project 3 – Single cell transcriptomics</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>

<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<link href="../assets/imglogo-basic-color-nobg-cmyk.svg" rel="icon" type="image/svg+xml">
<script src="../site_libs/quarto-html/quarto.js" type="module"></script>
<script src="../site_libs/quarto-html/tabsets/tabsets.js" type="module"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting-37eea08aefeeee20ff55810ff984fec1.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap-c90f6f86c3a7a5fe3a1fd67aae75894f.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light"><script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script><link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Source+Sans+Pro">
<!-- Matomo --><script>
  var _paq = window._paq = window._paq || [];
  /* tracker methods like "setCustomDimension" should be called before "trackPageView" */
  _paq.push(['trackPageView']);
  _paq.push(['enableLinkTracking']);
  (function() {
    var u="https://matomo.sib.swiss/";
    _paq.push(['setTrackerUrl', u+'matomo.php']);
    _paq.push(['setSiteId', '220']);
    var d=document, g=d.createElement('script'), s=d.getElementsByTagName('script')[0];
    g.async=true; g.src=u+'matomo.js'; s.parentNode.insertBefore(g,s);
  })();
</script><!-- End Matomo Code -->
</head>
<body class="nav-sidebar floating nav-fixed quarto-light">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top"><nav class="navbar navbar-expand-lg " data-bs-theme="dark"><div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="../index.html">
    <span class="navbar-title">Single cell transcriptomics</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
<li class="nav-item">
    <a class="nav-link active" href="../index.html" aria-current="page"> 
<span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../precourse_preparations.html"> 
<span class="menu-text">Precourse preparations</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../course_schedule.html"> 
<span class="menu-text">Course schedule</span></a>
  </li>  
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-day-1" role="link" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">day 1</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-day-1">
<li>
    <a class="dropdown-item" href="../day1/day1-1_setup.html">
 <span class="dropdown-text">Setup</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../day1/day1-1a_cellranger.html">
 <span class="dropdown-text">Introduction &amp; cellranger</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../day1/day1-2_analysis_tools_qc.html">
 <span class="dropdown-text">Analysis tools and QC</span></a>
  </li>  
    </ul>
</li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-day-2" role="link" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">day 2</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-day-2">
<li>
    <a class="dropdown-item" href="../day2/day2-1_normalization_scaling.html">
 <span class="dropdown-text">Normalization and scaling</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../day2/day2-2_dimensionality_reduction.html">
 <span class="dropdown-text">Dimensionality reduction</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../day2/day2-3_integration.html">
 <span class="dropdown-text">Integration</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../day2/day2-4_clustering.html">
 <span class="dropdown-text">Clustering</span></a>
  </li>  
    </ul>
</li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-day-3" role="link" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">day 3</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-day-3">
<li>
    <a class="dropdown-item" href="../day3/day3-1_cell_annotation.html">
 <span class="dropdown-text">Cell annotation</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../day3/day3-2_differential_gene_expression.html">
 <span class="dropdown-text">Differential gene expression</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../day3/day3-3_enrichment_analysis.html">
 <span class="dropdown-text">Enrichment analysis</span></a>
  </li>  
    </ul>
</li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-projects" role="link" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">projects</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-projects">
<li>
    <a class="dropdown-item" href="../projects/introduction.html">
 <span class="dropdown-text">Introduction</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../projects/project1.html">
 <span class="dropdown-text">Project 1</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../projects/project2.html">
 <span class="dropdown-text">Project 2</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../projects/project3.html">
 <span class="dropdown-text">Project 3</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../projects/project4.html">
 <span class="dropdown-text">Project 4</span></a>
  </li>  
    </ul>
</li>
</ul>
<ul class="navbar-nav navbar-nav-scroll ms-auto">
<li class="nav-item compact">
    <a class="nav-link" href="https://github.com/sib-swiss/single-cell-r-training/"> <i class="bi bi-github" role="img" aria-label="GitHub">
</i> 
<span class="menu-text"></span></a>
  </li>  
</ul>
</div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools">
</div>
      </div> <!-- /container-fluid -->
    </nav><nav class="quarto-secondary-nav"><div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item">Project 3</li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
    </div>
  </nav></header><!-- content --><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto"><div class="pt-lg-2 mt-2 text-left sidebar-header">
      <a href="../index.html" class="sidebar-logo-link">
      <img src="../assets/imglogo-basic-color-nobg-cmyk.svg" alt="" class="sidebar-logo py-0 d-lg-inline d-none"></a>
      </div>
</nav><div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active"><h2 id="toc-title">On this page</h2>
   
  <ul>
<li><a href="#available-data" id="toc-available-data" class="nav-link active" data-scroll-target="#available-data">Available data</a></li>
  <li>
<a href="#guidance-questions" id="toc-guidance-questions" class="nav-link" data-scroll-target="#guidance-questions">Guidance questions</a>
  <ul class="collapse">
<li><a href="#quality-control-qc" id="toc-quality-control-qc" class="nav-link" data-scroll-target="#quality-control-qc">1. Quality Control (QC)</a></li>
  <li><a href="#data-normalization-and-scaling" id="toc-data-normalization-and-scaling" class="nav-link" data-scroll-target="#data-normalization-and-scaling">2. Data Normalization and Scaling</a></li>
  <li><a href="#dimensionality-reduction" id="toc-dimensionality-reduction" class="nav-link" data-scroll-target="#dimensionality-reduction">3. Dimensionality Reduction</a></li>
  <li><a href="#unsupervised-clustering" id="toc-unsupervised-clustering" class="nav-link" data-scroll-target="#unsupervised-clustering">4. Unsupervised Clustering</a></li>
  <li><a href="#cluster-annotation" id="toc-cluster-annotation" class="nav-link" data-scroll-target="#cluster-annotation">5. Cluster Annotation</a></li>
  <li><a href="#differential-expression-de-and-ifn-i-response-analysis" id="toc-differential-expression-de-and-ifn-i-response-analysis" class="nav-link" data-scroll-target="#differential-expression-de-and-ifn-i-response-analysis">6. Differential Expression (DE) and IFN-I Response Analysis</a></li>
  <li><a href="#comparison-of-patient-groups" id="toc-comparison-of-patient-groups" class="nav-link" data-scroll-target="#comparison-of-patient-groups">7. Comparison of Patient Groups</a></li>
  <li><a href="#pathway-enrichment-analysis" id="toc-pathway-enrichment-analysis" class="nav-link" data-scroll-target="#pathway-enrichment-analysis">8. Pathway Enrichment Analysis</a></li>
  </ul>
</li>
  <li>
<a href="#additional-tips" id="toc-additional-tips" class="nav-link" data-scroll-target="#additional-tips">Additional tips</a>
  <ul class="collapse">
<li><a href="#key-figures-to-compare" id="toc-key-figures-to-compare" class="nav-link" data-scroll-target="#key-figures-to-compare">Key Figures to Compare</a></li>
  </ul>
</li>
  </ul></nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content"><header id="title-block-header" class="quarto-title-block default"><div class="quarto-title">
<h1 class="title">Project 3</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header><p>Project 3 focuses on understanding the role of type I interferon (IFN-I) responsiveness in shaping immune outcomes following PD1 blockade therapy; <a href="https://www.nature.com/articles/s41590-022-01262-7">paper</a>. Type I interferons are central regulators of anti-tumor immunity and responses to immunotherapy, but they also drive the feedback inhibition underlying therapeutic resistance. To investigate how pre-existing IFN-I responses influence therapeutic success, this study utilizes scRNA-seq data from <strong>healthy donors and 8 treated patients</strong> to analyze transcriptional responses in immune cells.</p>
<p>Unsupervised clustering of transcriptional profiles revealed distinct immune cell populations and IFN-I-induced responses. Differences in IFN-I responsiveness were linked to immune cell states and transcriptional programs that influence therapy outcomes. Patients with lower pre-therapy IFN-I responsiveness in CD4 and CD8 effector T cells (Teff cells) exhibited transcriptional signatures associated with improved immune function, whereas highly responsive Teff cells displayed gene expression patterns linked to immune dysfunction and therapy resistance.</p>
<p>Further analysis identified epigenetically imprinted IFN-I response states that predefine immune reactivity to therapy. Coexpression and network analyses demonstrated that IFN-I responsiveness influences functional T cell programs and systemic immune coordination. This study provides insights into how pre-existing immune states impact therapeutic success and highlights transcriptional markers that could be used to predict patient outcomes in PD1 blockade immunotherapy.</p>
<section id="available-data" class="level2"><h2 class="anchored" data-anchor-id="available-data">Available data</h2>
<p>Data has been downloaded and prepared for you from <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE199994">GEO GSE199994</a>.</p>
<p>In order to download the data, run:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">wget</span> https://single-cell-transcriptomics.s3.eu-central-1.amazonaws.com/projects/data/project3.tar.gz</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">tar</span> <span class="at">-xzvf</span> project3.tar.gz</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>After extracting, a directory <code>GSE199994</code> appears with the following format:</p>
<pre><code>GSE199994/
├── data
│&nbsp; &nbsp;├── HD1
│&nbsp; &nbsp;│&nbsp; &nbsp;├── barcodes.tsv.gz
│&nbsp; &nbsp;│&nbsp; &nbsp;├── features.tsv.gz
│&nbsp; &nbsp;│&nbsp; &nbsp;└── matrix.mtx.gz
│&nbsp; &nbsp;├── HD2
│&nbsp; &nbsp;│&nbsp; &nbsp;├── barcodes.tsv.gz
│&nbsp; &nbsp;│&nbsp; &nbsp;├── features.tsv.gz
│&nbsp; &nbsp;│&nbsp; &nbsp;└── matrix.mtx.gz
│&nbsp; &nbsp;├── P1
│&nbsp; &nbsp;│&nbsp; &nbsp;├── barcodes.tsv.gz
│&nbsp; &nbsp;│&nbsp; &nbsp;├── features.tsv.gz
│&nbsp; &nbsp;│&nbsp; &nbsp;└── matrix.mtx.gz
│&nbsp; &nbsp;├── P2
│&nbsp; &nbsp;│&nbsp; &nbsp;├── barcodes.tsv.gz
│&nbsp; &nbsp;│&nbsp; &nbsp;├── features.tsv.gz
│&nbsp; &nbsp;│&nbsp; &nbsp;└── matrix.mtx.gz
│&nbsp; &nbsp;├── P3
│&nbsp; &nbsp;│&nbsp; &nbsp;├── barcodes.tsv.gz
│&nbsp; &nbsp;│&nbsp; &nbsp;├── features.tsv.gz
│&nbsp; &nbsp;│&nbsp; &nbsp;└── matrix.mtx.gz
│&nbsp; &nbsp;├── P4
│&nbsp; &nbsp;│&nbsp; &nbsp;├── barcodes.tsv.gz
│&nbsp; &nbsp;│&nbsp; &nbsp;├── features.tsv.gz
│&nbsp; &nbsp;│&nbsp; &nbsp;└── matrix.mtx.gz
│&nbsp; &nbsp;├── P5
│&nbsp; &nbsp;│&nbsp; &nbsp;├── barcodes.tsv.gz
│&nbsp; &nbsp;│&nbsp; &nbsp;├── features.tsv.gz
│&nbsp; &nbsp;│&nbsp; &nbsp;└── matrix.mtx.gz
│&nbsp; &nbsp;├── P6
│&nbsp; &nbsp;│&nbsp; &nbsp;├── barcodes.tsv.gz
│&nbsp; &nbsp;│&nbsp; &nbsp;├── features.tsv.gz
│&nbsp; &nbsp;│&nbsp; &nbsp;└── matrix.mtx.gz
│&nbsp; &nbsp;├── P7
&nbsp; &nbsp;│&nbsp; &nbsp;├── barcodes.tsv.gz
&nbsp; &nbsp;│&nbsp; &nbsp;├── features.tsv.gz
&nbsp; &nbsp;│&nbsp; &nbsp;└── matrix.mtx.gz
&nbsp; &nbsp;└── P8
&nbsp; &nbsp; &nbsp; &nbsp;├── barcodes.tsv.gz
&nbsp; &nbsp; &nbsp; &nbsp;├── features.tsv.gz
&nbsp; &nbsp; &nbsp; &nbsp;└── matrix.mtx.gz
└── paper.pdf</code></pre>
<p>Showing us that we have two healthy donors (HD) and eight treated patients (P).</p>
<p>Now create a new project in the <code>project3</code> directory (Project (None) &gt; New Project …), and create a combined Seurat object from all the count matrices:</p>
<div class="cell">
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://satijalab.org/seurat">Seurat</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># vector of paths to all sample directories</span></span>
<span><span class="va">datadirs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.files.html">list.files</a></span><span class="op">(</span>path <span class="op">=</span> <span class="st">"project3/data"</span>, full.names <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># get the sample names and replace underscores with hyphens</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">datadirs</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/basename.html">basename</a></span><span class="op">(</span><span class="va">datadirs</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span><span class="op">(</span><span class="st">"_"</span>, <span class="st">"-"</span>, x <span class="op">=</span> <span class="va">_</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># create a large sparse matrix from all count data</span></span>
<span><span class="va">sparse_matrix</span> <span class="op">&lt;-</span> <span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.org/seurat/reference/Read10X.html">Read10X</a></span><span class="op">(</span>data.dir <span class="op">=</span> <span class="va">datadirs</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># create a seurat object from sparse matrix</span></span>
<span><span class="va">seu</span> <span class="op">&lt;-</span> <span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.github.io/seurat-object/reference/CreateSeuratObject.html">CreateSeuratObject</a></span><span class="op">(</span>counts <span class="op">=</span> <span class="va">sparse_matrix</span>,</span>
<span>project <span class="op">=</span> <span class="st">"InterferonStudy"</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="callout callout-style-default callout-important callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Project exercise
</div>
</div>
<div class="callout-body-container callout-body">
<p>With this dataset, go through the steps we have performed during the course, and try to reproduce the results provided in the paper.</p>
</div>
</div>
</section><section id="guidance-questions" class="level2"><h2 class="anchored" data-anchor-id="guidance-questions">Guidance questions</h2>
<p>If you’d like more structure, the following questions may guide your analyses:</p>
<section id="quality-control-qc" class="level3"><h3 class="anchored" data-anchor-id="quality-control-qc">1. Quality Control (QC)</h3>
<p>The paper’s methods for filtering were primarily based on mass cytometry (CyTOF) data. For your scRNA-seq data, you should apply standard QC metrics to filter out low-quality cells and genes using the <code>subset</code> function in Seurat.</p>
<ul>
<li>
<strong>Mitochondrial Genes:</strong> Use the pattern <code>"^MT-"</code> to calculate the percentage of mitochondrial reads (<code>PercentageFeatureSet</code>).</li>
<li>
<strong>Ribosomal Genes:</strong> Use the pattern <code>"^RP[SL]"</code> to calculate the percentage of ribosomal reads.</li>
<li>
<strong>Hemoglobin Genes:</strong> Use the pattern <code>"^HB[^(P)]"</code> to calculate the percentage of hemoglobin reads.</li>
<li>
<strong>Filtering:</strong> Use the <code>subset</code> function to filter cells based on the number of features (<code>nFeature_RNA</code>), UMI counts (<code>nCount_RNA</code>), and the calculated percentages of mitochondrial, ribosomal, and hemoglobin reads.</li>
</ul></section><section id="data-normalization-and-scaling" class="level3"><h3 class="anchored" data-anchor-id="data-normalization-and-scaling">2. Data Normalization and Scaling</h3>
<p>The paper used <code>arcsinh</code> transformation for their protein expression data. For your scRNA-seq data, you should use <code>SCTransform</code> to regress out unwanted variation.</p>
<ul>
<li>
<strong>Reproduce:</strong>
<ol type="1">
<li>Use the <code><a href="https://satijalab.org/seurat/reference/SCTransform.html">SCTransform()</a></code> function to normalize and scale the data and to regress out confounding factors like the number of UMIs (<code>nFeature_RNA</code>). The paper mentions using <code>scTransform</code> residuals for differential expression calculation, so this method is highly relevant.</li>
</ol>
</li>
</ul></section><section id="dimensionality-reduction" class="level3"><h3 class="anchored" data-anchor-id="dimensionality-reduction">3. Dimensionality Reduction</h3>
<p>The paper used <strong>UMAP</strong> for visualization. For this, you will first need to perform a linear dimensionality reduction.</p>
<ul>
<li>
<strong>Reproduce:</strong>
<ol type="1">
<li>Run PCA on the scaled data using the <code><a href="https://satijalab.org/seurat/reference/RunPCA.html">RunPCA()</a></code> function.</li>
<li>Use an Elbow Plot (<code><a href="https://satijalab.org/seurat/reference/ElbowPlot.html">ElbowPlot()</a></code>) to determine the appropriate number of principal components.</li>
<li>Run UMAP on the selected principal components using the <code><a href="https://satijalab.org/seurat/reference/RunUMAP.html">RunUMAP()</a></code> function to visualize the data in 2D.</li>
</ol>
</li>
</ul></section><section id="unsupervised-clustering" class="level3"><h3 class="anchored" data-anchor-id="unsupervised-clustering">4. Unsupervised Clustering</h3>
<p>The paper used the <strong>fast-PhenoGraph</strong> algorithm for clustering and identified <strong>39 distinct clusters</strong> in their <code>ex vivo</code> dataset.</p>
<ul>
<li>
<strong>Reproduce:</strong>
<ol type="1">
<li>Find the nearest neighbors using <code><a href="https://satijalab.org/seurat/reference/FindNeighbors.html">FindNeighbors()</a></code>.</li>
<li>Apply the clustering algorithm using <code><a href="https://satijalab.org/seurat/reference/FindClusters.html">FindClusters()</a></code>. You can adjust the <code>resolution</code> parameter to identify a similar number of clusters as reported in the paper. A resolution between 0.5 and 1.0 is a good starting point.</li>
</ol>
</li>
</ul></section><section id="cluster-annotation" class="level3"><h3 class="anchored" data-anchor-id="cluster-annotation">5. Cluster Annotation</h3>
<p>Annotate your clusters by identifying top marker genes for each cluster. The paper manually classified clusters based on lineage marker expression.</p>
<ul>
<li>
<strong>Reproduce:</strong>
<ol type="1">
<li>Use the <code><a href="https://satijalab.org/seurat/reference/FindAllMarkers.html">FindAllMarkers()</a></code> function to find marker genes for each cluster. You can set <code>min.pct = 0.25</code> and <code>logfc.threshold = 0.25</code> as a starting point.</li>
<li>Filter the marker genes (e.g., based on adjusted p-value) and use known canonical markers to assign cell type labels to your clusters.</li>
</ol>
</li>
</ul></section><section id="differential-expression-de-and-ifn-i-response-analysis" class="level3"><h3 class="anchored" data-anchor-id="differential-expression-de-and-ifn-i-response-analysis">6. Differential Expression (DE) and IFN-I Response Analysis</h3>
<p>A core part of the paper’s analysis is the <strong>IFN-I response capacity (IRC)</strong> score, which was based on the expression of a core set of IFN-I-stimulated proteins (ISPs). The paper also used the <strong>Wilcoxon’s rank-sum test</strong> for statistical testing.</p>
<ul>
<li>
<strong>IFN-I Response Score:</strong> To replicate this using gene expression data, you can use Seurat’s <code><a href="https://satijalab.org/seurat/reference/AddModuleScore.html">AddModuleScore()</a></code> function. The paper used the ISPs <strong>BST2, PKR, ISG15, MX1, IFIT3, and IRF7</strong>. You can use the corresponding genes to define your ISG gene list.</li>
<li>
<strong>Differential Expression:</strong> Use the <code><a href="https://satijalab.org/seurat/reference/FindMarkers.html">FindMarkers()</a></code> function to compare gene expression between groups (e.g., patients with high vs.&nbsp;low IFN-I responsiveness). You should specify <code>test.use = "wilcox"</code> for the Wilcoxon’s rank-sum test.</li>
</ul></section><section id="comparison-of-patient-groups" class="level3"><h3 class="anchored" data-anchor-id="comparison-of-patient-groups">7. Comparison of Patient Groups</h3>
<p>The paper’s key finding relates IFN-I responsiveness to therapy outcome.</p>
<ul>
<li>
<strong>Reproduce:</strong>
<ol type="1">
<li>After calculating the IFN-I response score, classify patient cells into high and low IFN-I responsive groups based on their scores. A common approach is to use the median or a quantile-based cutoff.</li>
<li>Perform a differential expression analysis between these two groups within specific cell clusters to identify gene expression patterns associated with improved or poor outcomes.</li>
</ol>
</li>
</ul></section><section id="pathway-enrichment-analysis" class="level3"><h3 class="anchored" data-anchor-id="pathway-enrichment-analysis">8. Pathway Enrichment Analysis</h3>
<p>The paper used <strong>Gene Set Enrichment Analysis (GSEA)</strong> and <strong>Ingenuity Pathway Analysis (IPA)</strong>. You can use the <code>R</code> package <code>clusterProfiler</code> for this task.</p>
<ul>
<li>
<strong>Reproduce:</strong>
<ol type="1">
<li>Take the list of differentially expressed genes from your DE analysis.</li>
<li>Perform a pathway enrichment analysis using <code>clusterProfiler</code> to identify affected biological pathways. You can use functions like <code>enrichGO</code> for Gene Ontology enrichment.</li>
</ol>
</li>
</ul></section></section><section id="additional-tips" class="level2"><h2 class="anchored" data-anchor-id="additional-tips">Additional tips</h2>
<section id="key-figures-to-compare" class="level3"><h3 class="anchored" data-anchor-id="key-figures-to-compare">Key Figures to Compare</h3>
<p>As you work through the analysis, you can compare your results to the following figures from the paper:</p>
<ul>
<li>
<strong>UMAP Plots:</strong> Compare your UMAP plots with <strong>Figure 1b</strong> and <strong>Figure 2a</strong> to check your clustering and visualization.</li>
<li>
<strong>IRC Score Plots:</strong> Compare the distributions of your calculated IFN-I response score across different cell populations to the violin plots in <strong>Figure 2d</strong>.</li>
</ul>


</section></section></main><!-- /main --><script id="quarto-html-after-body" type="application/javascript">
  window.document.addEventListener("DOMContentLoaded", function (event) {
    const icon = "";
    const anchorJS = new window.AnchorJS();
    anchorJS.options = {
      placement: 'right',
      icon: icon
    };
    anchorJS.add('.anchored');
    const isCodeAnnotation = (el) => {
      for (const clz of el.classList) {
        if (clz.startsWith('code-annotation-')) {                     
          return true;
        }
      }
      return false;
    }
    const onCopySuccess = function(e) {
      // button target
      const button = e.trigger;
      // don't keep focus
      button.blur();
      // flash "checked"
      button.classList.add('code-copy-button-checked');
      var currentTitle = button.getAttribute("title");
      button.setAttribute("title", "Copied!");
      let tooltip;
      if (window.bootstrap) {
        button.setAttribute("data-bs-toggle", "tooltip");
        button.setAttribute("data-bs-placement", "left");
        button.setAttribute("data-bs-title", "Copied!");
        tooltip = new bootstrap.Tooltip(button, 
          { trigger: "manual", 
            customClass: "code-copy-button-tooltip",
            offset: [0, -8]});
        tooltip.show();    
      }
      setTimeout(function() {
        if (tooltip) {
          tooltip.hide();
          button.removeAttribute("data-bs-title");
          button.removeAttribute("data-bs-toggle");
          button.removeAttribute("data-bs-placement");
        }
        button.setAttribute("title", currentTitle);
        button.classList.remove('code-copy-button-checked');
      }, 1000);
      // clear code selection
      e.clearSelection();
    }
    const getTextToCopy = function(trigger) {
        const codeEl = trigger.previousElementSibling.cloneNode(true);
        for (const childEl of codeEl.children) {
          if (isCodeAnnotation(childEl)) {
            childEl.remove();
          }
        }
        return codeEl.innerText;
    }
    const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
      text: getTextToCopy
    });
    clipboard.on('success', onCopySuccess);
    if (window.document.getElementById('quarto-embedded-source-code-modal')) {
      const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
        text: getTextToCopy,
        container: window.document.getElementById('quarto-embedded-source-code-modal')
      });
      clipboardModal.on('success', onCopySuccess);
    }
      var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp('/' + window.location.host + '/');
      var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
      }
      // Inspect non-navigation links and adorn them if external
     var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
      for (var i=0; i<links.length; i++) {
        const link = links[i];
        if (!isInternal(link.href)) {
          // undo the damage that might have been done by quarto-nav.js in the case of
          // links that we want to consider external
          if (link.dataset.originalHref !== undefined) {
            link.href = link.dataset.originalHref;
          }
        }
      }
    function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
      const config = {
        allowHTML: true,
        maxWidth: 500,
        delay: 100,
        arrow: false,
        appendTo: function(el) {
            return el.parentElement;
        },
        interactive: true,
        interactiveBorder: 10,
        theme: 'quarto',
        placement: 'bottom-start',
      };
      if (contentFn) {
        config.content = contentFn;
      }
      if (onTriggerFn) {
        config.onTrigger = onTriggerFn;
      }
      if (onUntriggerFn) {
        config.onUntrigger = onUntriggerFn;
      }
      window.tippy(el, config); 
    }
    const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
    for (var i=0; i<noterefs.length; i++) {
      const ref = noterefs[i];
      tippyHover(ref, function() {
        // use id or data attribute instead here
        let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
        try { href = new URL(href).hash; } catch {}
        const id = href.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note) {
          return note.innerHTML;
        } else {
          return "";
        }
      });
    }
    const xrefs = window.document.querySelectorAll('a.quarto-xref');
    const processXRef = (id, note) => {
      // Strip column container classes
      const stripColumnClz = (el) => {
        el.classList.remove("page-full", "page-columns");
        if (el.children) {
          for (const child of el.children) {
            stripColumnClz(child);
          }
        }
      }
      stripColumnClz(note)
      if (id === null || id.startsWith('sec-')) {
        // Special case sections, only their first couple elements
        const container = document.createElement("div");
        if (note.children && note.children.length > 2) {
          container.appendChild(note.children[0].cloneNode(true));
          for (let i = 1; i < note.children.length; i++) {
            const child = note.children[i];
            if (child.tagName === "P" && child.innerText === "") {
              continue;
            } else {
              container.appendChild(child.cloneNode(true));
              break;
            }
          }
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(container);
          }
          return container.innerHTML
        } else {
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(note);
          }
          return note.innerHTML;
        }
      } else {
        // Remove any anchor links if they are present
        const anchorLink = note.querySelector('a.anchorjs-link');
        if (anchorLink) {
          anchorLink.remove();
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        if (note.classList.contains("callout")) {
          return note.outerHTML;
        } else {
          return note.innerHTML;
        }
      }
    }
    for (var i=0; i<xrefs.length; i++) {
      const xref = xrefs[i];
      tippyHover(xref, undefined, function(instance) {
        instance.disable();
        let url = xref.getAttribute('href');
        let hash = undefined; 
        if (url.startsWith('#')) {
          hash = url;
        } else {
          try { hash = new URL(url).hash; } catch {}
        }
        if (hash) {
          const id = hash.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note !== null) {
            try {
              const html = processXRef(id, note.cloneNode(true));
              instance.setContent(html);
            } finally {
              instance.enable();
              instance.show();
            }
          } else {
            // See if we can fetch this
            fetch(url.split('#')[0])
            .then(res => res.text())
            .then(html => {
              const parser = new DOMParser();
              const htmlDoc = parser.parseFromString(html, "text/html");
              const note = htmlDoc.getElementById(id);
              if (note !== null) {
                const html = processXRef(id, note);
                instance.setContent(html);
              } 
            }).finally(() => {
              instance.enable();
              instance.show();
            });
          }
        } else {
          // See if we can fetch a full url (with no hash to target)
          // This is a special case and we should probably do some content thinning / targeting
          fetch(url)
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.querySelector('main.content');
            if (note !== null) {
              // This should only happen for chapter cross references
              // (since there is no id in the URL)
              // remove the first header
              if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
                note.children[0].remove();
              }
              const html = processXRef(null, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      }, function(instance) {
      });
    }
        let selectedAnnoteEl;
        const selectorForAnnotation = ( cell, annotation) => {
          let cellAttr = 'data-code-cell="' + cell + '"';
          let lineAttr = 'data-code-annotation="' +  annotation + '"';
          const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
          return selector;
        }
        const selectCodeLines = (annoteEl) => {
          const doc = window.document;
          const targetCell = annoteEl.getAttribute("data-target-cell");
          const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
          const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          const lines = annoteSpan.getAttribute("data-code-lines").split(",");
          const lineIds = lines.map((line) => {
            return targetCell + "-" + line;
          })
          let top = null;
          let height = null;
          let parent = null;
          if (lineIds.length > 0) {
              //compute the position of the single el (top and bottom and make a div)
              const el = window.document.getElementById(lineIds[0]);
              top = el.offsetTop;
              height = el.offsetHeight;
              parent = el.parentElement.parentElement;
            if (lineIds.length > 1) {
              const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
              const bottom = lastEl.offsetTop + lastEl.offsetHeight;
              height = bottom - top;
            }
            if (top !== null && height !== null && parent !== null) {
              // cook up a div (if necessary) and position it 
              let div = window.document.getElementById("code-annotation-line-highlight");
              if (div === null) {
                div = window.document.createElement("div");
                div.setAttribute("id", "code-annotation-line-highlight");
                div.style.position = 'absolute';
                parent.appendChild(div);
              }
              div.style.top = top - 2 + "px";
              div.style.height = height + 4 + "px";
              div.style.left = 0;
              let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
              if (gutterDiv === null) {
                gutterDiv = window.document.createElement("div");
                gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
                gutterDiv.style.position = 'absolute';
                const codeCell = window.document.getElementById(targetCell);
                const gutter = codeCell.querySelector('.code-annotation-gutter');
                gutter.appendChild(gutterDiv);
              }
              gutterDiv.style.top = top - 2 + "px";
              gutterDiv.style.height = height + 4 + "px";
            }
            selectedAnnoteEl = annoteEl;
          }
        };
        const unselectCodeLines = () => {
          const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
          elementsIds.forEach((elId) => {
            const div = window.document.getElementById(elId);
            if (div) {
              div.remove();
            }
          });
          selectedAnnoteEl = undefined;
        };
          // Handle positioning of the toggle
      window.addEventListener(
        "resize",
        throttle(() => {
          elRect = undefined;
          if (selectedAnnoteEl) {
            selectCodeLines(selectedAnnoteEl);
          }
        }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
        // Attach click handler to the DT
        const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
        for (const annoteDlNode of annoteDls) {
          annoteDlNode.addEventListener('click', (event) => {
            const clickedEl = event.target;
            if (clickedEl !== selectedAnnoteEl) {
              unselectCodeLines();
              const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
              if (activeEl) {
                activeEl.classList.remove('code-annotation-active');
              }
              selectCodeLines(clickedEl);
              clickedEl.classList.add('code-annotation-active');
            } else {
              // Unselect the line
              unselectCodeLines();
              clickedEl.classList.remove('code-annotation-active');
            }
          });
        }
    const findCites = (el) => {
      const parentEl = el.parentElement;
      if (parentEl) {
        const cites = parentEl.dataset.cites;
        if (cites) {
          return {
            el,
            cites: cites.split(' ')
          };
        } else {
          return findCites(el.parentElement)
        }
      } else {
        return undefined;
      }
    };
    var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
    for (var i=0; i<bibliorefs.length; i++) {
      const ref = bibliorefs[i];
      const citeInfo = findCites(ref);
      if (citeInfo) {
        tippyHover(citeInfo.el, function() {
          var popup = window.document.createElement('div');
          citeInfo.cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    }
  });
  </script>
</div> <!-- /content -->



</body></html>