---
title: "SIB course Single Cell Transcriptomics"
---

Single-cell RNA sequencing (scRNA-seq) allows researchers to study gene expression at the level of individual cells. This approach can, for example, help to identify different cell populations in a complex sample and describe their expression patterns. To generate and analyse scRNA-seq data, several methods are available, all with their strengths and weaknesses depending on the researchersâ€™ needs. This 3-day course will cover the main technologies as well as the main aspects to consider while designing an scRNA-seq experiment. In particular, it will combine the theoretical background of analytical methods with hands-on data analysis sessions focused on data generated by droplet-based platforms.

::: {.callout-caution}
## Looking for original material?

Are you looking for material of original SIB versions of the course? Find it at [sib-swiss.github.io/single-cell-r-training](https://sib-swiss.github.io/single-cell-r-training/)
:::

<figure>
    <img src="assets/images/umap_nonintegrated_dim50.png" width="500"/>
</figure>

## Teachers

- Michal Kolar [ORCiD](https://orcid.org/0000-0002-4593-1525)
- Lucie Pfeiferova [ORCiD](https://orcid.org/0000-0003-1089-0329)
- Jan Kubovciak [ORCiD](https://orcid.org/0000-0001-6446-9797)
- Yusuf Caglar Odabasi[ORCiD](https://orcid.org/0000-0003-0690-5575)

## Course support team
- Vojtech Melichar [ORCiD](https://orcid.org/0009-0000-7278-923X)
- Mathys Delattre
- Eva Rohlova

### Great thanks to
- Geert van Geest [ORCiD](https://orcid.org/0000-0002-1561-078X)
- Patricia Palagi [ORCiD](https://orcid.org/0000-0001-9062-6303)


## Attribution

We would like to thank Elixir for all the support and provided computational funds, also we would like to thank Metacentrum organization. This course is largely based on material from the SIB [sib-swiss.github.io/single-cell-r-training](https://sib-swiss.github.io/single-cell-r-training/), which we gratefully acknowledge. Parts of the original SIB Course were inspired by the [Broad Institute Single Cell Workshop](https://broadinstitute.github.io/2019_scWorkshop/index.html), the [CRUK CI Introduction to single-cell RNA-seq data analysis course](https://bioinformatics-core-shared-training.github.io/UnivCambridge_ScRnaSeq_Nov2021/) and courses previously developed by Walid Gharib at SIB. 

## License & copyright

**License:** [CC BY 4.0](https://github.com/sib-swiss/single-cell-training/blob/master/LICENSE.md)

**Copyright:** [SIB Swiss Institute of Bioinformatics](https://www.sib.swiss/)

## Learning outcomes

### General learning outcomes

By the end of the course, participants will possess the following abilities:

- Distinguish advantages and pitfalls of scRNA-seq.
- Design their own scRNA-seq experiment, by using common technologies like 10X Genomics.
- Apply quality control (QC) measures and utilise analysis tools to preprocess scRNA-seq data.
- Apply normalization, scaling, dimensionality reduction, and integration and clustering on sscRNA-seq data using R.
- Differentiate between cell annotation techniques to identify and characterize cell populations.
- Use differential gene expression analysis methods on scRNA-seq data to gain biological insights.
- Select enrichment analysis methods appropriate to the biological question and data.
- Develop a sscRNA-seq data analysis workflow from raw count matrix to differential gene expression with peer support and light guidance.

### Learning outcomes explained

To reach the general learning outcomes above, we have set a number of smaller learning outcomes. Each chapter starts with these smaller learning outcomes. Use these at the start of a chapter to get an idea what you will learn. Use them also at the end of a chapter to evaluate whether you have learned what you were expected to learn.

## Learning experiences

To reach the learning outcomes we will use lectures, exercises, polls and group work. During exercises, you are free to discuss with other participants. During lectures, focus on the lecture only.

### Exercises

Each block has practical work involved. Some more than others. The practicals are subdivided into chapters, and we'll have a (short) discussion after each chapter. All answers to the practicals are incorporated, but they are hidden. Do the exercise first by yourself, before checking out the answer. If your answer is different from the answer in the practicals, try to figure out why they are different.
