---
title: "Differential gene expression"
editor_options: 
  chunk_output_type: console
---


## Material

{{< downloadthis ../assets/pdf/202503_differential_analysis.pdf dname="DGE_and_enrichment_analysis" label="Download the presentation" icon="filetype-pdf" >}}

{{< video https://youtu.be/6Y5uCQWRRbg?si=mcV9qJUap1qvFjy3 >}}

-   More information on [pseudobulk analysis](https://bioconductor.org/books/3.17/OSCA.multisample/multi-sample-comparisons.html)
-   [Muscat](https://bioconductor.org/packages/release/bioc/html/muscat.html) for pseudobulk DGE.
-   [Paper](https://www.nature.com/articles/nmeth.4612) on the robustness of different differential expression analysis methods

## Exercises

### Find all markers for each cluster

Load the `seu` dataset you have created yesterday:

```{r}
#| warning: false
#| message: false
seu <- readRDS("day3/seu_day3-1.rds")
```

And load the following packages (install them if they are missing):

```{r}
#| warning: false
#| message: false
library(Seurat)
library(edgeR) # BiocManager::install("edgeR")
library(limma)
library(dplyr)
library(scuttle)
```

The function `FindAllMarkers` performs a Wilcoxon test to determine the genes differentially expressed between each cluster and the rest of the cells. Other types of tests than the Wilcoxon test are available. Check it out by running `?Seurat::FindAllMarkers`.

Now run analysis:

```{r}
#| output: false
de_genes <- Seurat::FindAllMarkers(seu,  min.pct = 0.25,
                                   only.pos = TRUE)
```

Subset the table to only keep the significant genes, and you can save it as a csv file if you wish to explore it further. Then extract the top 3 markers per cluster:

```{r}
de_genes <- subset(de_genes, de_genes$p_val_adj < 0.05)
write.csv(de_genes,
          "day3/de_genes_FindAllMarkers.csv",
          row.names = F, quote = F)

top_specific_markers <- de_genes %>%
  group_by(cluster) %>%
  top_n(3, avg_log2FC)
```

And generate e.g. a dotplot:

```{r}
dittoSeq::dittoDotPlot(seu,
                       vars = unique(top_specific_markers$gene), 
                       group.by = "RNA_snn_res.0.3")
```

::: callout-important
## Exercise

What are significant marker genes in cluster 0 and 8? Are the T cell genes in there?
:::

::: callout-hint
You can re-load the vector with immune genes with:

```{r}
tcell_genes <- c("IL7R", "LTB", "TRAC", "CD3D")
```
:::

::: {.callout-tip collapse="true"}
## Answer

```{r}
de_genes[de_genes$gene %in% tcell_genes,] |> knitr::kable()
```

So, yes, the T-cell genes are highly significant markers for cluster 0 and 8.
:::

### Differential expression between groups of cells

The `FindMarkers` function allows to test for differential gene expression analysis specifically between 2 groups of cells, i.e. perform pairwise comparisons, eg between cells of cluster 0 vs cluster 2, or between cells annotated as T-cells and B-cells.

First we can set the default cell identity to the cell types defined by `SingleR`:

```{r}
seu <- Seurat::SetIdent(seu, value = "SingleR_annot")
```

Run the differential gene expression analysis and subset the table to keep the significant genes:

```{r}
#| output: false
deg_cd8_cd4 <- Seurat::FindMarkers(seu,
                                   ident.1 = "CD8+ T cells",
                                   ident.2 = "CD4+ T cells",
                                   group.by = seu$SingleR_annot,
                                   test.use = "wilcox")
deg_cd8_cd4 <- subset(deg_cd8_cd4, deg_cd8_cd4$p_val_adj<0.05)
```

::: callout-important
## Exercise

Are CD8A, CD8B and CD4 in there? What does the sign (i.e. positive or negative) mean in the log fold change values? Are they according to the CD8+ and CD4+ annotations? Check your answer by generating a violin plot of a top differentially expressed gene.
:::

::: {.callout-tip collapse="true"}
## Answer

You can check out the results with:

```{r}
#| eval: false
View(deg_cd8_cd4)
```

```{r}
#| echo: false
knitr::kable(deg_cd8_cd4)
```

For an explanation of the log fold change have a look at `?Seurat::FindMarkers`. At **Value** it says:

> `avg_logFC`: log fold-chage of the average expression between the two groups. Positive values indicate that the gene is more highly expressed in the first group

To view CD8A, CD8B and CD4:

```{r}
deg_cd8_cd4[c("CD4", "CD8A", "CD8B"),]
```

Indeed, because we compared ident.1 = "CD8+ T cells" to ident.2 = "CD4+ T cells", a negative log2FC for the CD4 gene indicates a lower expression in CD8+ T-cells than in CD4+ T-cells, while a positive log2FC for the CD8A and CD8B genes indicates a higher expression in CD8+ T-cells.

Plotting the genes in these two T-cell groups only:

```{r}
Seurat::VlnPlot(seu, 
            features = c("CD4", "CD8A", "CD8B"),
            idents = c("CD8+ T cells", "CD4+ T cells"))
```
:::

### Differential expression using `limma`

The Wilcoxon test implemented in `FindMarkers` does not allow you to test for complex design (eg factorial experiments) or to include batch as a covariate. It doesn't allow you to run paired-sample T tests for example.

For more complex designs, we can use `edgeR` or `limma` which are designed for microarray or bulk RNA seq data and provide a design matrix that includes covariates for example, or sample IDs for paired analyses.

We will load an object containing only pro B cells, both from healthy tissues (PBMMC), and malignant tissues (ETV6-RUNX1).

::: callout-warning
Please NOTE that in the original design of this data set, the healthy and malignant tissues were not patient-matched, i.e. the real design was not the one of paired healthy and malignant tissues. However, for demonstration purposes, we will show you how to run a paired analysis, and do as if the PBMMC-1 and ETV6-RUNX1-1 samples both came from the same patient 1, the PBMMC-2 and ETV6-RUNX1-2 samples both came from the same patient 2, etc...
:::

We can load the object and explore its UMAP and meta.data like this:

```{r}
proB <- readRDS("course_data/proB.rds")

Seurat::DimPlot(proB, group.by = "orig.ident")

table(proB@meta.data$type)

head(proB@meta.data)
```

::: callout-note
If you want to know how this pro-B cell subset is generated, have a look at the script [here](https://raw.githubusercontent.com/sib-swiss/single-cell-training/master/scripts/generate_object_proB.R).
:::


Let's have a look at the UMAP (again), coloured by celltype:

```{r}
Seurat::DimPlot(proB, group.by = "type")
```

Let's say we are specifically interested to test for differential gene expression between the tumor and normal samples.

::: callout-note
Here we could also test for e.g. healthy versus diseased within a celltype/cluster.
:::

Now we will run differential expression analysis between tumor and healthy cells using the patient ID as a covariate by using `limma`.

Prepare the pseudobulk count matrix:

```{r}
#| output: false
#taking the proB data 
Seurat::DefaultAssay(proB) <- "RNA"
Seurat::Idents(proB) <- proB$orig.ident

## add the patient id also for paired DGE
proB$patient.id<-gsub("ETV6-RUNX1", "ETV6_RUNX1", proB$orig.ident)
proB$patient.id<-sapply(strsplit(proB$patient.id, "-"), '[', 2)

## Here we do perform pseudo-bulk:
##first a mandatory column of sample needs to be added to the meta data that is the grouping factor, should be the samples
proB$sample <- factor(proB$orig.ident)

# aggergate the cells per sampple
bulk <- Seurat::AggregateExpression(proB, group.by = "sample",
                                    return.seurat = TRUE,
                                    assay = "RNA")

# create a metadata data frame based on the aggregated cells
meta_data <- unique(proB@meta.data[, c("orig.ident",
                                            "sample", "type",
                                            "patient.id")])
rownames(meta_data) <- meta_data$orig.ident
bulk@meta.data <- meta_data[colnames(bulk), ]

```

```{r}
##have a look at the counts
counts <- Seurat::GetAssayData(bulk, layer = "counts") |> as.matrix()

head(counts)

#have a look at the colData of our new object summed, can you see type and 
#patient.id are there
head(bulk@meta.data)
```

Generate a `DGEList` object to use as input for `limma` and filter the genes to remove lowly expressed genes. How many are left?

```{r}
#As in the standard limma analysis generate a DGE object

y <- edgeR::DGEList(counts, samples = bulk@meta.data)

##filter lowly expressed (recommanded for limma)
keep <- edgeR::filterByExpr(y, group = bulk$type)
y <- y[keep,]

##see how many genes were kept 
summary(keep)
```

Generate a design matrix, including patient ID to model for a paired analysis. If you need help to generate a design matrix, check out the very nice [edgeR User Guide](https://bioconductor.org/packages/release/bioc/vignettes/edgeR/inst/doc/edgeRUsersGuide.pdf), sections 3.3 and 3.4. Extract the sample ID from the meta.data, then create the design matrix:

```{r}
## Create the design matrix and include the technology as a covariate:
design <- model.matrix(~0 + y$samples$type + y$samples$patient.id)

# Have a look
design

# change column/rownames names to more simple group names: 
colnames(design) <- make.names(c("ETV6-RUNX1", "PBMMC","patient2","patient3"))
rownames(design) <- rownames(y$samples)
```

Specify which contrast to analyse:

```{r}
contrast.mat <- limma::makeContrasts(ETV6.RUNX1 - PBMMC,
                                     levels = design)

```

Firt, we perform TMM normalization using edgeR, and then `limma` can perform the transformation with `voom`, fit the model, compute the contrasts and compute test statistics with `eBayes`:

```{r}
dge <- edgeR::calcNormFactors(y)  

#Do limma
vm <- limma::voom(dge, design = design)
fit <- limma::lmFit(vm, design = design)
fit.contrasts <- limma::contrasts.fit(fit, contrast.mat)
fit.contrasts <- limma::eBayes(fit.contrasts)
```

We can use `topTable` to get the most significantly differentially expressed genes, and save the full DE results to an object. How many genes are significant? Are you suprised by this number?

```{r}
# Show the top differentially expressed genes:
limma::topTable(fit.contrasts, number = 10, sort.by = "P")
limma_de <- limma::topTable(fit.contrasts, number = Inf, sort.by = "P")
length(which(limma_de$adj.P.Val<0.05))
```

And we can check whether this corresponds to the counts by generating a violin plot, or a gene downregulated in tumor, or a gene upregulated in tumor:

```{r}
#| message: false
Seurat::VlnPlot(proB, "S100A9", split.by = "type")
Seurat::VlnPlot(proB, "SOCS2", split.by = "type")
```

## Some QC plots

### Histogram of p-values
```{r}
library(ggplot2)

ggplot(limma_de, aes(x = adj.P.Val)) +
  geom_histogram(binwidth = 0.01, fill = "skyblue", color = "black") +
  theme_minimal() +
  labs(
    title = "Histogram of Adjusted P-values",
    x = "Adjusted P-value",
    y = "Frequency (Number of Genes)"
  ) +
  geom_vline(xintercept = 0.05, linetype = "dashed", color = "red") +
  annotate("text", x = 0.07, y = max(hist(limma_de$adj.P.Val, plot = FALSE)$counts) * 0.9,
           label = "FDR = 0.05", color = "red", hjust = 0)
```

A "healthy" histogram is L-shaped, with a high bar at 0 and then relatively flat. If your histogram looks like that, your statistical testing has likely performed well. Read more about distribution of histogram p-value: [How to interpret a p-value histogram](http://varianceexplained.org/statistics/interpreting-pvalue-histogram/)


### MA plot
```{r}
sig_threshold_adj_p <- 0.05
sig_threshold_logFC <- 1
limma_de_ma_plot <- limma_de %>%
  mutate(
    is_significant = case_when(
      adj.P.Val < sig_threshold_adj_p & logFC > sig_threshold_logFC ~ "Upregulated",
      adj.P.Val < sig_threshold_adj_p & logFC < -sig_threshold_logFC ~ "Downregulated",
      TRUE ~ "Not Significant"
    ),
    gene_label = ifelse(abs(logFC) > 5 | (adj.P.Val < 1e-5 & abs(logFC) > 3), rownames(limma_de), NA)
  )

ggplot(limma_de_ma_plot, aes(x = AveExpr, y = logFC, color = is_significant)) +
  geom_point(alpha = 0.6, size = 1) + 
  scale_color_manual(values = c("Upregulated" = "red", "Downregulated" = "blue", "Not Significant" = "grey")) +
  theme_minimal() +
  labs(
    title = "MA Plot",
    x = "Average Expression (log2)",
    y = "Log2 Fold Change",
    color = "Significance"
  ) +
  geom_hline(yintercept = c(-sig_threshold_logFC, sig_threshold_logFC), linetype = "dashed", color = "darkgreen") +
  geom_hline(yintercept = 0, linetype = "solid", color = "black") +
  geom_text(aes(label = gene_label), na.rm = TRUE, size = 3, hjust = -0.1, vjust = 0.5, check_overlap = TRUE)
```

A "healthy" MA plot is symmetrical around the logFC = 0 line, with most points tightly clustered near zero logFC at high average expression (AveExpr), and a wider spread at low average expression. This indicates no systematic biases and effective detection of differential expression.


## Differential analysis using `Seurat`
We can run a similar analysis with `Seurat`, but this will not take into account the paired design. Run the code below.

```{r}
tum_vs_norm <- Seurat::FindMarkers(proB, 
                                   ident.1 = "ETV6-RUNX1", 
                                   ident.2 = "PBMMC", 
                                   group.by = "type")
tum_vs_norm_s <- subset(tum_vs_norm, tum_vs_norm$p_val_adj<0.05)
```

::: callout-important
## Exercise (extra)

How many genes are significant? How does the fold change of these genes compare to the fold change of the top genes found by limma?
:::

::: {.callout-tip collapse="true"}
## Answer

```{r}
dim(tum_vs_norm_s) 
```
We find 3820 significant genes. 
:::

::: callout-important
## Exercise

Can you now try to make histogram of p-values from `Seurat` analysis?
:::

::: {.callout-tip collapse="true"}
## Answer

```{r}
library(ggplot2)

ggplot(tum_vs_norm, aes(x = p_val_adj)) +
  geom_histogram(binwidth = 0.01, fill = "skyblue", color = "black") +
  theme_minimal() +
  labs(
    title = "Histogram of Adjusted P-values",
    x = "Adjusted P-value",
    y = "Frequency (Number of Genes)"
  ) +
  geom_vline(xintercept = 0.05, linetype = "dashed", color = "red") +
  annotate("text", x = 0.07, y = max(hist(tum_vs_norm$p_val_adj, plot = FALSE)$counts) * 0.9,
           label = "FDR = 0.05", color = "red", hjust = 0)

```


Many differential expression tests (including those implemented in Seurat, like `Wilcoxon Rank Sum test`) will not compute a p-value for genes that are expressed in too few cells in one or both of the groups being compared.
:::



## Comparison of results between `Seurat` and `limma`

If we merge the `FindMarkers` and the `limma` results, keep `limma`'s most significant genes and plot:

```{r}
merge_limma_FindMarkers <- merge(tum_vs_norm_s, limma_de, by="row.names",
                           all.x=T)

par(mar=c(4,4,4,4))
plot(merge_limma_FindMarkers$avg_log2FC,
    merge_limma_FindMarkers$logFC,
    xlab="log2FC Wilcoxon", ylab="log2FC limma",
    pch=15, cex=0.5)
abline(a=0, b=1, col="red")
```


::: callout-warning
## Keep the object

Keep the `tum_vs_norm` and `limma_de` objects because we will use this output later for the enrichment analysis in the next section.
:::

```{r}
#| echo: false

saveRDS(tum_vs_norm, "day3/tum_vs_norm_day3-2.rds")
saveRDS(limma_de, "day3/limma_de_day3-2.rds")
saveRDS(proB, "day3/proB_day3-2.rds")
```
